lang/src/org/partiql/lang/eval/NaturalExprValueComparators.kt

```diff
@@ -180,6 +180,7 @@ enum class NaturalExprValueComparators(private val nullOrder: NullOrder) : Compa
                     lVal.isPosInf && rVal.isPosInf -> return EQUAL
                     lVal.isPosInf -> return MORE
                     rVal.isPosInf -> return LESS
+                    lVal.isZero() && rVal.isZero() -> 0 // for negative zero
```

Comment:
As is, the branch does not return and I think the value of `0` is lost (not kept in a variable or returned). 

Also, does the spec state what PartiQL should do for comparing representations of `0` that are in different types, e.g., `0`.compareTo(`-0e0`)? 

If we are to treat all values that denote 0 and -0 as EQUAL then line 183 should `return EQUAL`, if not, I would expect us to take the absolute value (drop the -) and then delegate to the types implementation for comparison.

Reply:
Fixed. AFAIK PartiQL equality uses the mathematical definition, e.g. `0`, `-0`, `0.000`, `-0.0`, `0e0`, `0.000000000` are all equal